library/network/src/modules/SuSEFirewall.rb

```diff
@@ -405,6 +405,23 @@ def GetAllKnownInterfaces
       deep_copy(known_interfaces)
     end
 
+    # Function returns list of all known interfaces.
+    #
+    # @return	[Array<String>] of interfaces
+    # @example GetListOfKnownInterfaces() -> ["eth1", "eth2", "modem0", "dsl5"]
+    def GetListOfKnownInterfaces
+      interfaces = []
+
+      Builtins.foreach(GetAllKnownInterfaces()) do |interface_map|
+        interfaces = Builtins.add(
+          interfaces,
+          Ops.get_string(interface_map, "id", "")
+        )
+      end
+
+      deep_copy(interfaces)
```

Comment:
Given that we know that the called method cannot return `nil` and the hashes always do contain an id, this can be greatly simplified :)
```rb
GetAllKnownInterfaces().map { |i| i["id"] }
```